lib/libc/zoneinfo/mk_latest_tzrule.py

```diff
@@ -0,0 +1,208 @@
+#typecnt[4]
+#charcnt[4]
+#header size = 44
+
+#for i : timecnt
+# at (int) or (long long)
+#for i : timecnt
+# typ (char) 
+#for i : typecnt
+# gmtoff(int)
+# isdst(char)
+# abbrind(char)
+#for i : charcnt
+# chars(char)
+#for i : leapcnt
+# trans(int) or (long long)
+# corr(int)
+#for i : typecnt
+# if(ttisstd != 0)
+# ttisstd(char)
+#for i : typecnt
+# if(ttisgmt != 0)
+# ttisgmt(char)
+
+MAGIC_SIZE    = 4
+VERSION_SIZE  = 1
+RESERVED_SIZE = 15
+TTISGMT_SIZE  = 4
+TTISSTD_SIZE  = 4
+LEAPCNT_SIZE  = 4
+TIMECNT_SIZE  = 4
+TYPECNT_SIZE  = 4
+CHARCNT_SIZE  = 4
+INT_SIZE      = 4
+CHAR_SIZE     = 1
+
+def latest_rule(file_name):
+    now = int(time.time())
+    fp = open(file_name, "rb")
+    data = fp.read()
+    fp.close()
+
+    fp = open(file_name, "wb")
+
+    store_size = [4, 8]
+    point = 0
+
+    for size in store_size:
+        magic = data[point:point + MAGIC_SIZE]
+        fp.write(magic)
+        point += MAGIC_SIZE
+        version = data[point:point + VERSION_SIZE]
+        fp.write(version)
+        point += VERSION_SIZE
+        reserved = data[point:point + RESERVED_SIZE]
+        fp.write(reserved)
+        point += RESERVED_SIZE
+
+        ttisgmt = struct.unpack('>I', data[point:point + TTISGMT_SIZE])[0]
+        fp.write(struct.pack('>I', ttisgmt))
+        point += TTISGMT_SIZE
+
+        ttisstd = struct.unpack('>I', data[point:point + TTISSTD_SIZE])[0]
+        fp.write(struct.pack('>I', ttisstd))
+        point += TTISSTD_SIZE
+
+        leapcnt = struct.unpack('>I', data[point:point + LEAPCNT_SIZE])[0]
+        fp.write(struct.pack('>I', leapcnt))
+        point += LEAPCNT_SIZE
+
+        timecnt = struct.unpack('>I', data[point:point + TIMECNT_SIZE])[0]
+        point += TIMECNT_SIZE
+
+        typecnt = struct.unpack('>I', data[point:point + TYPECNT_SIZE])[0]
+        point += TYPECNT_SIZE
+
+        charcnt = struct.unpack('>I', data[point:point + CHARCNT_SIZE])[0]
+        point += CHARCNT_SIZE
+
+        at_list = list()
+        index = 0
+        for i in range(0, timecnt):
+            if size == 4 :
+                at = struct.unpack('>i', data[point:point + size])[0]
+            else :
+                at = struct.unpack('>q', data[point:point + size])[0]
+
+            if at >= now and index == 0:
+                index = i - 1
+
+            point += size
+            at_list.append(at)
+
+        if index == 0 :
+            index = timecnt - 1
+
+        if index < 0:
+            index = 0
+
+        fp.write(struct.pack('>I', timecnt - index))
+        fp.write(struct.pack('>I', typecnt))
+        fp.write(struct.pack('>I', charcnt))
+
+        for i in range(index, timecnt) :
+            if size == 4 :
+                fp.write(struct.pack('>i', at_list[i]))
+            else :
+                fp.write(struct.pack('>q', at_list[i]))
+
+        typ_list = list()
+        for i in range(0, timecnt):
+            typ = struct.unpack('b', data[point:point + CHAR_SIZE])[0]
+            point += CHAR_SIZE
```

Comment:
How about using `sys.getsizeof`?

Reply:
I wanted to unite with a variable of length that should be read.
`
point:point + CHAR_SIZE
point += CHAR_SIZE
`